functional_test.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+set +ex
+
+if [ "$#" -lt 1 ]; then
+    echo "Usage: functional_test.sh TAG"
+    exit 1
+fi
+
+TAG=$1
+
+TMPDIR=$(mktemp -d)
+
+# To run this script locally, you may need to run `docker pull $tag` from the model repo,
+# where your credentials will be populated.
+DOCKER_IMAGE_TAG="us.gcr.io/model-159019/gh:$TAG"
+
+# Import data into graphhopper's internal format
+docker run \
+    -v "$(pwd)/web/test-data/:/graphhopper/test-data/" \
+    -v "$TMPDIR:/graphhopper/transit_data/"\
+    --rm \
+     "$DOCKER_IMAGE_TAG" \
+     java -Xmx2g -Xms1g -XX:+UseG1GC -XX:MetaspaceSize=100M \
+     -Ddw.graphhopper.datareader.file=test-data/kansas-city-extract-mini.osm.pbf \
+     -Ddw.graphhopper.gtfs.file=test-data/mini_kc_gtfs.tar \
+     -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar -server com.graphhopper.http.GraphHopperApplication import default_gh_config.yaml
+
+# Run link-mapping step
+docker run \
+    -v "$TMPDIR:/graphhopper/transit_data/"\
+    --rm \
+    "$DOCKER_IMAGE_TAG" \
+    java -Xmx2g -Xms1g -XX:+UseG1GC -XX:MetaspaceSize=100M \
+    -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar com.graphhopper.http.GraphHopperApplication gtfs_links default_gh_config.yaml
+
+# Run server in background
+docker run --rm  --name functional_test_server -p 50051:50051 -p 8998:8998 -v "$TMPDIR:/graphhopper/transit_data/" \
+    "$DOCKER_IMAGE_TAG" &
+
+echo "Waiting for graphhopper server to start up"
+sleep 30
```

Comment:
sleep is fine (how long does this mini container take in practice?)...

but I wonder if we could ping until available to speed things up? also not sure the variability we might experience

Reply:
Yup, I don't love needing to use a blunt instrument like this, but I really wasn't sure how best to ping the container, and I figured that this would be ~good enough for the time being. In practice, the container seems to be ready in more like 10-15 secs locally, but I wanted to be a little conservative to avoid false positive failures.